src/feditest/nodedrivers/mastodon/api.py

```diff
@@ -0,0 +1,295 @@
+"""
+Mastodon API Node
+
+See MastodonNode for a description of the supported/required node parameters.
+"""
+
+import time
+from typing import Any
+from urllib.parse import urlparse
+
+import httpx
+from mastodon import Mastodon
+
+from feditest import nodedriver
+from feditest.protocols import Node, NodeDriver
+from feditest.protocols.activitypub import ActivityPubNode
+
+
+def _dereference(uri: str) -> dict:
+    return httpx.get(
+        uri,
+        follow_redirects=True,
+        verify=False,
+        headers={"Accept": "application/activity+json"},
+    ).json()
+
+
+class Actor:
+    """Encapsulates account/actor related information. For a Mastodon "actor"
+    it contains a reference to an API wrapper and the account data. For an actor
+    without a Mastodon access_token, it has the uri and the AP actor document.
+    """
+
+    def __init__(self, role: str, hostname: str, params: str):
+        if access_token := params.get("access_token"):
+            self.api = Mastodon(
+                api_base_url=f"https://{hostname}", access_token=access_token
+            )
+            self.data = self.api.account_verify_credentials()
+        else:
+            self.api = None
+            self.data = None
+        self.role = role
+        self.uri = params["uri"]
+        self.actor = _dereference(self.uri)
+
+    @property
+    def is_mastodon(self):
+        return self.api is not None
+
+
+class MastodonNode(ActivityPubNode):
```

Comment:
I think we want the `MastodonNode` to be a subclass of `FediverseNode` and then the `MastodonUbosNode` to be a sublcass of `MastodonNode`. That allows us to use the same `MastodonNode` code in a UBOS context and in any other context like as a Saas scenario.
(But then, no need to do that right now, you did mark it experimental)

Reply:
It was that way originally, but it didn't make sense to me since that only adds the WebFinger Node interface, which didn't appear necessary since we already have WebFinger tests without it.

Reply:
> I think we want the `MastodonNode` to be a subclass of `FediverseNode` and then the `MastodonUbosNode` to be a sublcass of `MastodonNode`. That allows us to use the same `MastodonNode` code in a UBOS context and in any other context like as a Saas scenario. (But then, no need to do that right now, you did mark it experimental)

Did you want this done before merging the PR? If so, I prefer to withdraw the PR until I have time to do the redesign.